app/query/fetch/api.ts

```diff
@@ -66,6 +66,22 @@ export const useFetchArtistTopSongs = () => {
   }
 }
 
+export const useFetchArtistAllSongs = () => {
+  const client = useClient()
+
+  return async (id: string) => {
+    const artistResponse = await client().getArtist({ id })
+
+    let songs: Song[] = [];
+    for (const album of artistResponse.data.albums) {
+      const albumResponse = await fetchAlbum(album.id, client())
```

Comment:
The chained fetches here aren't ideal, and with React Query I think we'd generally want to spread these out into separate queries, but since there isn't any support for adding songs to the queue as they come in yet I think this is fine.

Although, I do think it would make sense to parallelize the subsequent `fetchAlbum` calls instead of waiting on each one before starting the next.

Reply:
Got it. I'm working on an alternate implementation that uses parallel queries to fetch the albums. It also resuse the Artist query instead of always fetching.

The only part I haven't figured out is how to delay these queries until the user clicks the button. Right now they are firing upon the Artist page loading.

Reply:
You can set `enabled: false` in the query options, and then either use some local state to control `enabled` by the button press or I think you might be able to call the `refetch()` function that the query hook returns as well.